parity/run.rs

```diff
@@ -194,7 +194,7 @@ pub fn execute(cmd: RunCmd, logger: Arc<RotatingLogger>) -> Result<(), String> {
 	let mut sync_config = SyncConfig::default();
 	sync_config.network_id = match cmd.network_id {
 		Some(id) => id,
-		None => spec.network_id(),
+		None => spec.network_id() as usize,
```

Comment:
possibility of overflow? changing the sync service to use a fixed bit-width seems sensible regardless.